wled00/network.cpp

```diff
@@ -178,12 +184,21 @@ void WiFiEvent(WiFiEvent_t event)
     case SYSTEM_EVENT_ETH_START:
       DEBUG_PRINTLN(F("ETH Started"));
       break;
+    case SYSTEM_EVENT_ETH_GOT_IP:
+      if (Network.isEthernet()) {
```

Comment:
Why do you need to check if this is Ethernet? After all, this is Ethernet event.

On a side note there is a bug in network.cpp dependency.
```c++
bool NetworkClass::isConnected()
{
#if defined(ARDUINO_ARCH_ESP32) && defined(WLED_USE_ETHERNET)
  return (WiFi.localIP()[0] != 0 && WiFi.status() == WL_CONNECTED) || (ETH.localIP()[0] != 0 && ETH.linkUp());
#else
  return (WiFi.localIP()[0] != 0 && WiFi.status() == WL_CONNECTED);
#endif
}

bool NetworkClass::isEthernet()
{
#if defined(ARDUINO_ARCH_ESP32) && defined(WLED_USE_ETHERNET)
  return (ETH.localIP()[0] != 0) && WiFi.getMode() == WIFI_MODE_NULL;
#endif
  return false;
}
```
This is the correct implementation if you care to add it to this PR.
@softhack007 has asked about this bug.